target/linux/ramips/dts/mt7620a_trendnet_tha103ac.dts

```diff
@@ -0,0 +1,192 @@
+#include "mt7620a.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/leds/common.h>
+
+/ {
+	compatible = "trendnet,tha103ac", "ralink,mt7620a-soc";
+	model = "TRENDNET THA-103AC (EU)";
+
+	aliases {
+		label-mac-device = &ethernet;
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+	};
+
+	keys: keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 9 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio0 13 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+
+		leds {
+		compatible = "gpio-leds";
+
+		led_power: power {
+			function = LED_FUNCTION_POWER;
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&gpio0 14 GPIO_ACTIVE_LOW>;
+		};
+
+		wps {
+			function = LED_FUNCTION_WPS;
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&gpio0 10 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan {
+			function = LED_FUNCTION_WLAN;
+			color = <LED_COLOR_ID_BLUE>;
+			gpios = <&gpio1 14 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan_25_blue {
+			label = "blue:wlan25";
+			color = <LED_COLOR_ID_BLUE>;
+			gpios = <&gpio2 16 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan_50_blue {
+			label = "blue:wlan50";
+			color = <LED_COLOR_ID_BLUE>;
+			gpios = <&gpio2 17 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan_75_blue {
+			label = "blue:wlan75";
+			color = <LED_COLOR_ID_BLUE>;
+			gpios = <&gpio2 18 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan_100_blue {
+			label = "blue:wlan100";
+			color = <LED_COLOR_ID_BLUE>;
+			gpios = <&gpio2 19 GPIO_ACTIVE_LOW>;
+		};
```

Comment:
Would be nice to sort the leds by their gpio numbers. Why not to add default triggers to some of them according to their functions?

Reply:
Sorted and added RSSI defaults.

Reply:
Thanks. Is it possible use `function` (new style) for all leds?

An example:
```
		led-8 {
			color = <LED_COLOR_ID_BLUE>;
			/* LED_FUNCTION_RSSIHIGH isn't implemented yet */
			function = "rssihigh";
			gpios = <&gpio2 19 GPIO_ACTIVE_LOW>;
		};
```

`01_leds` should be updated in case of the change.

Reply:
> unction =

Can you assist with more guidance on how 01_leds should be adjusted i.r.o. ?

Also re: commend "/* LED_FUNCTION_RSSIHIGH isn't implemented yet */", not sure what to do, just leave the comments in?

I can't seems to find an example of RRSI indicator using FUNCTION.

Reply:
> Can you assist with more guidance on how 01_leds should be adjusted i.r.o. ?

I checked your `01_leds` and think that it doesn't require any adjustments because you're already use `color:function` format.

> Also re: commend "/* LED_FUNCTION_RSSIHIGH isn't implemented yet */", not sure what to do, just leave the comments in?

Yes. Additionally would be nice to send upstream patch to implement these functions in new Linux kernels. If you has such experience or would like to get it.

> I can't seems to find an example of RRSI indicator using FUNCTION.

You will be the first. 😉